moto/ram/models.py

```diff
@@ -111,7 +115,6 @@ def add_principals(self, principals: List[str]) -> None:
                     f"Principal ID {principal} is malformed. Verify the ID and try again."
                 )
 
-        for principal in principals:
```

Comment:
I think this should stay, actually, despite the performance improvement that it would give, for the simple reason that we only want to add the principals if all of them are valid. 

If a user provides two principals, one valid and the second invalid, with this change the first principal will be added, but the request will fail on the second one. For the user, it would be a bit confusing if some of the principals were successfully added, despite the fact that the request failed.